x/src/cargo.rs

```diff
@@ -5,29 +5,26 @@ use crate::{utils::project_root, Result};
 use anyhow::anyhow;
 use log::{info, warn};
 use std::{
-    collections::HashMap,
     ffi::{OsStr, OsString},
     path::Path,
     process::{Command, Output, Stdio},
     time::Instant,
 };
 
-pub struct Cargo {
+pub struct Cargo<'a> {
     inner: Command,
-    direct_args: Vec<OsString>,
     pass_through_args: Vec<OsString>,
-    env_additions: HashMap<OsString, OsString>,
+    env_additions: &'a [(&'a OsStr, &'a OsStr)],
```

Comment:
You probably want to enable user's to push env on one at a time or in batches (it makes the API a little nicer) so if you really want to hold onto the env vars then you should change this to take ownership, ie have this be of type `Vec<(OsString, OsString)>`

Reply:
But it looks like you decided (based on the below) that you're not logging the env vars so maybe drop this entirely?

Reply:
ah, thanks -- this has sunk in, at least a bit.

Reply:
done.